head/src/main/kotlin/io/qalipsis/core/head/report/DataSeriesServiceImpl.kt

```diff
@@ -196,4 +204,47 @@ internal class DataSeriesServiceImpl(
         )
     }
 
+    override suspend fun refresh() {
+        backgroundScope.launch {
+            var pageable = Pageable.from(0, SIZE)
+            var pagedDataSeries: Page<DataSeriesEntity>
+            do {
+                pagedDataSeries = dataSeriesRepository.findAll(pageable)
+                handleQueryUpdate(pagedDataSeries)
+                pageable = pagedDataSeries.nextPageable()
+            }
+            while (pagedDataSeries.pageNumber < pagedDataSeries.totalPages - 1)
+        }
+    }
+
+    /**
+     * Handle the update of each query for a given batch of data series entity.
+     */
+    private suspend fun handleQueryUpdate(pagedDataSeries: Page<DataSeriesEntity>) {
+        try {
+            val countOfUpdatedQueries = pagedDataSeries.numberOfElements
+            logger.trace { "Updating the queries of ${pagedDataSeries.numberOfElements} retrieved data series" }
+            val dataSeriesEntities = pagedDataSeries.content
+            dataSeriesEntities.forEach { dataSeriesEntity ->
+                val tenant = tenantRepository.findReferenceById(dataSeriesEntity.tenantId)
```

Comment:
Just an idea. What if you would order the data series by their tenants, so that you can fetch the tenants only once, and not for each individual data series?